code/go/0chain.net/smartcontract/dbs/event/transaction.go

```diff
@@ -51,3 +51,10 @@ func (edb *EventDb) GetTransactionByBlockHash(blockHash string, offset, limit in
 	res := edb.Store.Get().Model(Transaction{}).Where(Transaction{BlockHash: blockHash}).Offset(offset).Limit(limit).Scan(&tr)
 	return tr, res.Error
 }
+
+// GetTransactionByBlockHashes returns all transactions for a given block hashes
+func (edb *EventDb) GetTransactionByBlockHashes(blockHashes []string, offset, limit int) ([]Transaction, error) {
+	tr := []Transaction{}
+	res := edb.Store.Get().Model(Transaction{}).Where("block_hash IN (?)", blockHashes).Offset(offset).Limit(limit).Scan(&tr)
```

Comment:
use single query with join to Block table

Reply:
We'd better group the transactions results with block hash, otherwise the caller would not know what block those transactions belong to.

Reply:
@peterlimg  Do you mean group by block number? I am confused because we are returning slice of txns and each txn has a block_hash field which can be used to determine to which block each txn belong.

Reply:
Ah, didn't realize the transaction in event db has block hash field. Should be fine to keep it as it is.